plugin/src/test/kotlin/com/ritense/plugin/service/PluginServiceTest.kt

```diff
@@ -355,6 +356,95 @@ internal class PluginServiceTest {
         }
     }
 
+    @Test
+    fun `should invoke delegateTask method`(){
+        val task = mock<DelegateTask>()
+        val execution = mock<DelegateExecution>()
+        val processLink = PluginProcessLink(
+            PluginProcessLinkId.newId(),
+            "process",
+            "activity",
+            Mapper.INSTANCE.get().readTree("{\"test\":123}") as ObjectNode,
+            PluginConfigurationId.newId(),
+            "test-action-task"
+        )
+
+        val pluginDefinition = newPluginDefinition(TestPlugin::class.java.name)
+        val pluginConfiguration = newPluginConfiguration(pluginDefinition)
+        val testDependency = mock<TestDependency>()
+
+        whenever(pluginConfigurationRepository.getById(any())).thenReturn(pluginConfiguration)
+        whenever(pluginFactory.canCreate(any())).thenReturn(true)
+        whenever(pluginFactory.create(any())).thenReturn(TestPlugin(testDependency))
+        whenever(task.processInstanceId).thenReturn("test")
+        whenever(task.execution).thenReturn(execution)
+        whenever(execution.processInstanceId).thenReturn("test")
+        whenever(valueResolverService.resolveValues(any(), any(), any())).thenReturn(mapOf("test" to 123))
+
+        pluginService.invoke(task, processLink)
+
+        verify(testDependency).processInt(123)
+    }
+
+    @Test
+    fun `should throw exception when invoking delegateTask method with resolved variable where result does not match argument type`(){
+        val task = mock<DelegateTask>()
```

Comment:
Know that camunda mockito provides a fake implementation for these delegates:
https://github.com/camunda-community-hub/camunda-platform-7-mockito#delegatetaskexecutionfake